pilot/pkg/xds/ads.go

```diff
@@ -480,6 +495,23 @@ func (s *DiscoveryServer) initConnection(node *core.Node, con *Connection) error
 	return nil
 }
 
+func authorizeConnection(con *Connection) error {
+	for _, rawID := range con.Identities {
+		spiffeID, err := spiffe.ParseIdentity(rawID)
+		if err != nil {
+			continue
+		}
+		if con.proxy.ConfigNamespace != "" && spiffeID.Namespace != con.proxy.ConfigNamespace {
```

Comment:
in what case the ConfigNamespace is empty? is that considered an error or what configs does istiod return if it's not considered an error?

Reply:
I would say: if ConfigNamespace is empty, populate it from the identity. 

This simplifies the client code and config - in particular for VMs, one less thing to configure.

Reply:
In a normal cluster, never. It seems its not considered an error today. Based on a quick look through the code, seems like its looking things up in the `""` namespace which obviously doesn't exist - so I guess it will get no special configs. Note that this still means we get all the mesh wide stuff like services, endpoints, etc. Its probably not a bad idea to just reject this

Reply:
I was thinking that a connection without any node metadata - but with proper identity - should work. It's simpler and cleaner. 
Why should we have to pass the namespace and service account - extracted with complicated syntax - when it's already in the cert ( and will be in the JWT as well ) ?

Reply:
No problem with rejecting if they're different - but we should stop requiring all the work to get something we can get from the identity, it's waste.

Reply:
We will need it for telemetry in the proxy anyways. I am fine with defaulting though, but we should definitely reject if they don't match IMO or we may end up with hard to debug errors